docs/ilab-model-engine.md

```diff
@@ -0,0 +1,204 @@
+        new `ilab data generate`
+
+```
+
+So in `ilab` the clear source of information is taxonomy, or the new skills/knowledge brought into generate. This data then goes through processes that end up in the sink, which is a running/actionable process which the user can interact with. Providing commands that clearly display these different process is key.  groups for these larger "buckets" of processes makes sense as well due to the likelihood of more commands being needed to properly interact with the newly generated model.
+
+
+### Alternatives
+
+An alternative flow to `ilab` -> `child-command` -> `grandchild-command` is:
+
+`ilab` -> `verb` -> `noun`. The positive to this approach is commands would sound better: `ilab generate data`. However there are a few negatives. While this is easier to say, it makes less sense from an organizational standpoint:
+
+```console
+ilab
+|
+|_______download__|
+|_______convert___|-----model
+|_______train_____|
+```
+
+This structure works when commands are ALL linked to the same `verb`. What happens when a group like `download` gets a command that the others don't? Then it looks more like this:
+
+```console
+ilab
+|
+|_______download
+|       |____model
+|       |____taxonomy
+|_______convert
+|       |_____model
+|_______train
+|       |_____model
+```
+
+This is opposed to the structure in this EP, which might have a duplicate here and there for something like `download`, but in this structure duplicate commands are the norm and will result in a larger and clunkier codebase that is confusing to read.
+
+This first structure looks nice, and `ilab download model` sounds nice. However in terms of implementation, this makes little sense. in `click`, the cli library we use, `ilab` is a "group" that commands are currently "grouped under". In this Model, `ilab` would be a group, `download`, `convert` etc would be a sub-group, and `model` would be a command under ALL of these groups. This would require an implementation of the `model` function in different packages all linking to a different group parent. From the perspective of user contributions and general code readability doesn't make much sense. Part of this design is to encourage user contributions by making the structure of the codebase logical. While these commands make the commands sound better, they make them harder to group, and understand.
+
+### General workflow as compared to alternatives
+
+how to use `ilab` with this new structure:
+
+1. `ilab config init`
+2. `ilab model download`
+3. `ilab data generate`
+4. `ilab model train --convert`
+5. `ilab model serve -i` (or followed by `ilab model chat`)
+
+currently one has to:
+
+1. `ilab init`
+2. `ilab download`
+3. `ilab serve` 
+4. `ilab generate`
+5. `ilab train`
+7. `ilab convert`
+8. `ilab serve --model XXX`
```

Comment:
Like above swap and first state current flow and then the nerw proposed improved flow?